lib/ui/form/form_screen.dart

```diff
@@ -58,14 +64,26 @@ class FormScreenState extends ConsumerState<FormScreen> {
 
   @override
   Widget build(BuildContext context) {
+    ref.listen<FormState>(formViewModelProvider, (_, formState) {
+      formState.maybeWhen(
+        surveySubmissionSuccess: (outroMessage) =>
+            _navigateToSurveySuccessScreen(outroMessage),
+        orElse: () {},
+      );
+    });
+
     final surveyDetails = ref.watch(_surveyDetailsStreamProvider).value;
     final errorMessage = ref.watch(_errorStreamProvider).value ?? "";
+
     return ref.watch(formViewModelProvider).when(
           init: () => _buildFormScreen(isLoading: true),
           loading: () => _buildFormScreen(isLoading: true),
           loadSurveyDetailsSuccess: () => _buildFormScreen(
             surveyDetails: surveyDetails,
           ),
+          surveySubmissionSuccess: (_) {
+            return Container();
```

Comment:
Flutter offers this for empty view instead

```suggestion
            return const SizedBox.shrink();
```